src/main/java/io/github/thebusybiscuit/slimefun4/core/attributes/ExternallyInteractable.java

```diff
@@ -0,0 +1,116 @@
+package io.github.thebusybiscuit.slimefun4.core.attributes;
+
+import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItem;
+import org.bukkit.Location;
+import org.bukkit.inventory.ItemStack;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Implement this interface for any {@link SlimefunItem} to provide methods for
+ * external code to 'interact' with the item when placed as a block in the world.
+ *
+ * @author Sefiraat
+ *
+ */
+public interface ExternallyInteractable {
+
+    /**
+     * This method should be used by the implementing class to fulfill the actions needed
+     * when being interacted with returning the result of the interaction.
+     *
+     * @param location
+     *                 The {@link Location} of the Block being targeted for the interaction.
+     *
+     * @return The {@link InteractionResult} denoting the result of the interaction.
+     */
+    @Nonnull InteractionResult onInteract(@Nonnull Location location);
+
+    /**
+     * This class represents the result of an interaction on an {@link ExternallyInteractable} item.
+     */
+    class InteractionResult {
+
+        private final boolean interactionSuccessful;
+        private @Nullable String resultMessage;
```

Comment:
Use case?

Reply:
It just allows the object being interacted with to define a string message - could be anything. Then it's up to the interacting class how/if it deal withs it. 
So for example in my Light Controller block, it would just say something like 'Failed due to not having enough power' or 'Failed due to missing connection' and it could then be fed back to the user from the class that called for the interaction